utils/python/CIME/test_scheduler.py

```diff
@@ -148,13 +148,17 @@ def __init__(self, test_names, test_data=None,
 
         # Setup phases
         self._phases = list(PHASES)
+        # These will mark the -no phase as PENDING in TestStatus 
         if self._no_setup:
-            self._phases.remove(SETUP_PHASE)
+            i = self._phases.index(SETUP_PHASE)
```

Comment:
INCOMPLETE_PHASE will be inserted multiple times if no_build or no_setup is True. Is that what you intended? I think this code will work due to the check you put in producer, but it leaves the self._phases list in a somewhat odd state. I can accept this as-is but I might see if I can refactor this later.